prov/efa/src/dgram/efa_ep.c

```diff
@@ -642,10 +453,14 @@ int efa_ep_open(struct fid_domain *domain_fid, struct fi_info *user_info,
 			return ret;
 	}
 
-	ep = efa_ep_alloc(user_info);
+	ep = calloc(1, sizeof(*ep));
 	if (!ep)
 		return -FI_ENOMEM;
 
+	ret = efa_base_ep_construct(&ep->base_ep, user_info);
+	if (ret)
+		goto err_ep_destroy;
+
 	ret = ofi_endpoint_init(domain_fid, &efa_util_prov, user_info, &ep->base_ep.util_ep,
```

Comment:
The initialization of `base_ep.util_ep` should also go into `efa_base_ep_construct`

Reply:
Need to pass the context, util_prov and the progress function as a function pointer to `efa_base_ep_construct`. I didn't think that was worth the complexity.

Reply:
I see. We should do that after we merge the `efa_util_prov` and `rxr_prov_util`. Can you add a TODO note here?

Reply:
done